internal/tools/pipeline/spec/pipeline.go

```diff
@@ -105,9 +105,34 @@ func (p *Pipeline) GetRunUserID() string {
 	return mustUserID(userID)
 }
 
-// GetUserID first to get run userID,if not exists, to get submit userID
+func (p *Pipeline) GetOwnerUserID() string {
+	var userID string
+	if p.Extra.OwnerUser != nil && p.Extra.OwnerUser.ID != nil {
+		userID = fmt.Sprintf("%v", p.Extra.OwnerUser.ID)
+	}
+	return mustUserID(userID)
+}
+
+// GetExecuteUserID get userID to execute pipeline
+// permissions for the task belong to the executor userID
+// first to get owner userID,if not exists, to get run userID
+func (p *Pipeline) GetExecuteUserID() string {
```

Comment:
confused. what's difference between RunUser and ExecuteUser?
changed to `GetOwnerOrRunUserID`

Reply:
adjusted